pkg/ccl/changefeedccl/changefeed_processors.go

```diff
@@ -574,12 +576,15 @@ func (ca *changeAggregator) noteResolvedSpan(resolved jobspb.ResolvedSpan) error
 		canCheckpointSpans(&ca.flowCtx.Cfg.Settings.SV, ca.lastFlush)
 
 	if checkpointFrontier || checkpointSpans {
-		defer func() {
-			ca.lastFlush = timeutil.Now()
-		}()
+		ca.lastFlush = timeutil.Now()
+		ca.flushSkipped = false
```

Comment:
Is there a benefit to changing `lastFlush` to mean "the start of the last flush attempt" as opposed to "the last completion of a flush"?

Reply:
Ohh... That's what it was :)
Just didn't want to keep defer.... I can bring it back.

Reply:
Reverted to using defer.